react/components/BuyButton/index.js

```diff
@@ -45,44 +42,60 @@ export class BuyButton extends Component {
   }
 
   handleAddToCart = async () => {
-    const { skuItems, isOneClickBuy, orderFormContext, push } = this.props
+    const { addToCart, skuItems, isOneClickBuy } = this.props
     this.setState({ isAddingToCart: true })
 
-    const variables = {
-      items: skuItems.map(skuItem => {
-        const { skuId } = skuItem
+    if (isOneClickBuy) location.assign(CONSTANTS.CHECKOUT_URL)
+
+    let toastMessage = null
+    try {
+      const minicartItems = skuItems.map(skuItem => {
+        const {
+          skuId: id,
+          variant: skuName,
+          price: sellingPrice,
+          detailUrl,
+          imageUrl,
+          quantity,
+          seller,
+          name,
+          options,
+          listPrice,
+        } = skuItem
+
         return {
-          id: parseInt(skuId),
+          detailUrl,
+          id,
+          imageUrl,
+          listPrice,
+          name,
+          quantity,
+          sellingPrice,
+          skuName,
+          seller,
+          options,
           index: 1,
-          ...pick(['quantity', 'seller', 'options'], skuItem),
         }
-      }),
-    }
+      })
 
-    variables.orderFormId = orderFormContext.orderForm.orderFormId
+      const {
+        data: { addToCart: linkStateItems },
+      } = await addToCart(minicartItems)
 
-    if (isOneClickBuy) location.assign(CONSTANTS.CHECKOUT_URL)
+      const success = skuItems.map(skuItem =>
+        find(linkStateItems, { id: skuItem.skuId })
+      )
 
-    push({
-      event: 'addToCart',
-      items: skuItems,
-    })
+      toastMessage = () => this.toastMessage(success.length >= 1)
+    } catch (err) {
+      console.error(err)
+      toastMessage = () => this.toastMessage(false)
+    }
 
-    await orderFormContext.addItem({ variables }).then(
-      mutationRes => {
-        const { items } = mutationRes.data.addItem
-        const success = skuItems.map(skuItem =>
-          find(items, { id: skuItem.skuId })
-        )
-
-        orderFormContext.refetch()
-        this.toastMessage(success.length >= 1)
-      },
-      () => {
-        this.toastMessage(false)
-      }
+    setTimeout(
+      () => this.setState({ isAddingToCart: false }, toastMessage),
+      5e2
```

Comment:
Some reason for that?

Reply:
The 5e2 in this case

Reply:
@brunojdo didn't like the UX because you don't see the buy-button spinner. So, I added a timeout just to show that.

Reply:
Why not just 500? I'm sure that the majority of JS developers wouldn't know at first how many milliseconds `5e2` represents

Reply:
That's true. I'll change it (and I promise I won't use this argument against ramda 🙊)